src/Agent.Listener/Configuration/ConfigurationManager.cs

```diff
@@ -486,23 +492,60 @@ private void ConfigureAutoLogonIfNeeded(CommandSettings command)
                     var shallRestart = command.GetRestartPermission();
                     if(shallRestart)
                     {
-                        Trace.Info("Restarting the machine now");
+                        Trace.Info("Restarting the machine in 5 seconds");
                         _term.WriteLine(StringUtil.Loc("RestartIn5SecMessage"));
                         Process.Start("shutdown.exe", "-r -t 5");
                     }
+                    else
+                    {
+                        Trace.Info("No restart happened. As the interactive session is configured for a different user agent will not be launched");
+                    }
+                    return;
                 }
-                else
+
+                var startupProcessPath = Path.Combine(HostContext.GetDirectory(WellKnownDirectory.Bin), "agentservice.exe");
+                Trace.Info("Launching the agent process.");
+                _term.WriteLine(StringUtil.Loc("AgentLaunch"));
+                
+                var processInfo = new ProcessStartInfo();
+                processInfo.UseShellExecute = false;
+                processInfo.CreateNoWindow = true;
+                processInfo.FileName = startupProcessPath;
+                processInfo.Arguments = "runasprocess";                
+
+                var agentServiceProcess = Process.Start(processInfo);
+                Trace.Verbose($"Started the agentservice process. Id - {agentServiceProcess.Id}");
+            }
+            catch(Exception ex)
+            {
+                Trace.Error(ex);
+                _term.WriteLine(StringUtil.Loc("AutoLogonConfigurationFailureMessage"));
+            }
+        }
+
+        private void UnConfigureAutoLogonIfNeeded()
+        {
+            try
+            {
+                var iConfigManager = HostContext.GetService<IInteractiveSessionConfigurationManager>();
+                if(iConfigManager.IsInteractiveSessionConfigured())
                 {
-                    var startupProcessPath = Path.Combine(HostContext.GetDirectory(WellKnownDirectory.Bin), "agentservice.exe");                    
-                    Trace.Info("Launching the agent process.");
-                    _term.WriteLine(StringUtil.Loc("AgentLaunch"));
-                    Process.Start(startupProcessPath, "runasprocess");
+                    Trace.Verbose("Interactive session was not configured on the agent. Returning.");
+                    return;
+                }                
+
+                AssertAdminAccess();
+                var agentServiceProcess = Process.GetProcessesByName("agentservice");
+                if(agentServiceProcess.Length > 0)
+                {
+                    agentServiceProcess[0].Kill(); 
```

Comment:
changed the approach.

---
In reply to: [116709694](https://github.com/Microsoft/vsts-agent/pull/965#discussion_r116709694) [](ancestors = 116709694)